hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -695,6 +704,7 @@ public void setAsMaster() {
         setJoined();
```

Comment:
well spotted; actually `changeClusterVersion` kicks-off the cluster state change to apply cluster version on each member of the cluster. I think this is not necessary at this point, as it is enough to just set the local cluster version property, if not already set; subsequent `FinalizeJoinOperation`s will apply the cluster version to each member joining the cluster.